aten/src/ATen/cuda/CUDASolver.cpp

```diff
@@ -1173,6 +1240,155 @@ void xpotrs(
   TORCH_CUSOLVER_CHECK(cusolverDnXpotrs(handle, params, uplo, n, nrhs, dataTypeA, A, lda, dataTypeB, B, ldb, info));
 }
 
+template <>
+void xgeqrf_bufferSize<float>(CUDASOLVER_XGEQRF_BUFFERSIZE_ARGTYPES(float)) {
+  TORCH_CUSOLVER_CHECK(cusolverDnXgeqrf_bufferSize(
+      handle,
+      params,
+      m,
+      n,
+      CUDA_R_32F,
```

Comment:
Would it be beneficial to get rid of the template specialization for 64-bit API with something like this? 
https://github.com/pytorch/pytorch/blob/bcdcf347cb0b2998dca866c1409bf0f0ca371550/aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu#L492-L506

Reply:
Currently, it doesn't bring any value and it's a matter of taste, I think we can continue using templates here as `xgeqrf` supports only the same combination of data types as the original `geqrf`.
![image](https://user-images.githubusercontent.com/19621411/115234513-c79ed500-a121-11eb-979a-424720609ca2.png)